### ISSUE TITLE
vault list format not valid yaml / json response

### ISSUE THREAD

**Issue by loeffert** [hashicorp/vault#1213]
I've added a new entry to vault and then deleted it.
But the folder seems still to exist.

If I do a list on that folder now, I get the output:
"No entries found"

Which is fine so far. But if I request a yaml or a json I do not get a valid one back.

`vault list --format json path/to/an/empty/folder`
`vault list --format yaml path/to/an/empty/folder`

It also returns the message "No entries found" instead of an empty json / yaml.

**Comment by jefferai** [comment]
Hi @loeffert ,

You'll see "No entries found" regardless of whether or not you have written a folder in the past. You can easily reproduce this by starting up a `-dev` server and running the `list` command. If you check with `curl` you can see the expected output:
```
$ curl -X LIST -H "X-Vault-Token: $(cat ~/.vault-token)" http://127.0.0.1:8200/v1/secret/foo
{"lease_id":"","renewable":false,"lease_duration":0,"data":{"keys":["bar"]},"warnings":null,"auth":null}
$ vault delete secret/foo/bar
Success! Deleted 'secret/foo/bar'
$ curl -X LIST -H "X-Vault-Token: $(cat ~/.vault-token)" http://127.0.0.1:8200/v1/secret/foo
{"lease_id":"","renewable":false,"lease_duration":0,"data":{},"warnings":null,"auth":null}
$ curl -X LIST -H "X-Vault-Token: $(cat ~/.vault-token)" http://127.0.0.1:8200/v1/secret
{"lease_id":"","renewable":false,"lease_duration":0,"data":{},"warnings":null,"auth":null}
```
No output is produced in this case regardless of format, that is, `stdout` remains empty. What you're seeing is what's being printed to `stderr`.

If you want to submit a PR that will output an empty YAML.JSON object instead of nothing when no data is returned, I'm happy to look at it.

**Comment by jefferai** [comment]
This should be fixed in 0.9.5 or so.
